checker.c

```diff
@@ -1,21 +1,48 @@
 #include <stdio.h>
-#include <assert.h>
+#include "checker.h"
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+float Temperature;
+float StateOfCharge;
+float ChargeRate;
+
+int IsNotWithinLimit(float currentValue, float min, float max)
+{
+    return ((currentValue < min) || (currentValue > max));
+}
+
+int isTemperatureOkay(void)
+{
+    if(IS_TEMPERATURE_NOT_IN_LIMIT(Temperature))
+    {
+        printf("Temperature out of range!\n");
+        return 0;
```

Comment:
You have `#define` for IN_LIMIT and NOT_IN_LIMIT. That is a better practice than using numbers like 0 and 1.
However, you have also used 0 and 1. Use the named constants in all places.

Reply:
yes, sure.